lib/ansible/plugins/connection/psrp.py

```diff
@@ -522,6 +531,13 @@ def _build_kwargs(self):
             max_envelope_size=self._psrp_max_envelope_size,
             operation_timeout=self._psrp_operation_timeout,
         )
+
+        # Check if PSRP version supports newer read_timeout argument (needs pypsrp 0.3.0+)
+        if hasattr(pypsrp, 'FEATURES') and 'wsman_read_timeout' in pypsrp.FEATURES:
+            self._psrp_conn_kwargs['read_timeout'] = self._psrp_read_timeout
+        elif self._psrp_read_timeout:
+            display.debug("Installed pypsrp version does not support 'read_timeout', using 'connection_timeout' instead.")
```

Comment:
Right now read_timeout will always be set as there's a default value. You either need to remove this or remove the default value on the read_timeout doc entry.

Reply:
I wonder if there's a way to access the original values, before defaults were applied.
I can remove the debug-statement too, however I think it should be a warning instead.

Reply:
I agree that this should be a warning, if the user has explicitly set this and they are on an older version they should know why it isn't applying.

Reply:
For reconnection retries we also do display.debug(), so I will change these into a warning() too.

Reply:
Sounds good to me

Reply:
The only problem is that we cannot detect if the value was a default value, or if it was set by the user. I think that's why we have it as debug().

Reply:
Hmm that is true, we will probably have think this through a bit more. It sounds like we can either keep it as is or not have a default option and warn the user if it is set and an older pypsrp version is installed. Not having a default option defined in the config allows us to take advantage of the defaults in pypsrp if it is undefined. We can still tell if the user has explicitly set it to null or has left it undefined and fire the warning accordingly. What do you think?

Reply:
Fine by me.